File: src/sdk-reference/cpp/1/search-result/snippets/search-result.cpp
try {
  for (int i = 0; i < 5; i++) {
    kuzzle->document->create("nyc-open-data", "yellow-taxi", "", R"({
      "category": "suv"
    })");
  }
  for (int i = 5; i < 15; i++) {
    kuzzle->document->create("nyc-open-data", "yellow-taxi", "", R"({
      "category": "limousine"
    })");
  }
  kuzzle->index->refresh("nyc-open-data");

  kuzzleio::query_options options = {};
  options.scroll = "1m";
  options.size = 2;

  kuzzleio::SearchResult* response = kuzzle->document->search(
    "nyc-open-data",
    "yellow-taxi",
    R"({
      "query": {
        "match": {
          "category": "suv"
        }
      }
    })",
    &options);

  kuzzleio::SearchResult* next_page = response->next();

  std::cout << "Successfully retrieved " << next_page->fetched << " documents" << std::endl;
} catch (kuzzleio::KuzzleException e) {
  std::cerr << e.getMessage() << std::endl;
}

File: src/sdk-reference/csharp/1/search-result/snippets/search-result.cs
try {
  for (int i = 0; i < 5; i++) {
    kuzzle.document.create("nyc-open-data", "yellow-taxi", "", @"{
      ""category"": ""suv""
    }");
  }
  for (int i = 5; i < 15; i++) {
    kuzzle.document.create("nyc-open-data", "yellow-taxi", "", @"{
      ""category"": ""limousine""
    }");
  }
  kuzzle.index.refresh("nyc-open-data");

  query_options options = {};
  options.scroll = "1m";
  options.size = 2;

  SearchResult response = kuzzle.document.search(
    "nyc-open-data",
    "yellow-taxi",
    @"{
      ""query"": {
        ""match"": {
          ""category"": ""suv""
        }
      }
    }",
    options);

  SearchResult next_page = response.next();

  Console.WriteLine("Successfully retrieved " + next_page.fetched + " documents");
} catch (KuzzleException e) {
  Console.Error.WriteLine(e.getMessage());
}